charts/argocd-apps/templates/application.yaml

```diff
@@ -0,0 +1,55 @@
+{{- range .Values.applications }}
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: {{ .name }}
+  namespace: cluster-services
+  finalizers:
+  # Deleting this Application will delete its resources (may not be appropriate)
+  - resources-finalizer.argocd.argoproj.io
+spec:
+  project: govuk
+  source:
+    repoURL: 'git@github.com/alphagov/govuk-helm-charts'
+    path: "{{ .chartPath | default (printf "charts/%s" .name) }}"
+    targetRevision: {{ .targetRevision | default "HEAD" }}
+
+    # Helm overrides specific to environment
+    # Same as setting through values.yaml, but these take precedence.
+{{ if .helm }}
+    helm:
+      {{ if .helm.parameters }}
+      parameters:
+        {{- .helm.parameters | toYaml | trim | nindent 6 }}
+      {{- end }}
+{{- end }}
+
+  destination:
+    server: 'https://kubernetes.default.svc'
+    namespace: apps
+  syncPolicy:
+    automated: # automated sync by default retries failed attempts 5 times with following delays between attempts ( 5s, 10s, 20s, 40s, 80s ); retry controlled using `retry` field.
+      prune: true # Specifies if resources should be pruned during auto-syncing ( false by default ).
+      selfHeal: true # Specifies if partial app sync should be executed when resources are changed only in target Kubernetes cluster and no git change detected ( false by default ).
+      allowEmpty: false # Allows deleting all application resources during automatic syncing ( false by default ).
+    syncOptions:     # Sync options which modifies sync behavior
+    - Validate=false # disables resource validation (equivalent to 'kubectl apply --validate=false') ( true by default ).
```

Comment:
What's the rationale for disabling resource validation? Basic pre-apply validation seems like a good thing with little to no downsides, unless there's some issue caused elsewhere with Argo or Helm?

Reply:
These options were included in the default application.yaml file provided in the docs (@sengi linked to the relevant section in the docs which explains why: https://github.com/alphagov/govuk-helm-charts/pull/13#discussion_r731677751). I've removed this config as it isn't currently required.